pkg/reconciler/route/resources/ingress_test.go

```diff
@@ -1045,6 +1045,78 @@ func TestMakeIngressWithTLS(t *testing.T) {
 	}
 }
 
+func TestMakeIngressWithHTTPOption(t *testing.T) {
+	targets := map[string]traffic.RevisionTargets{}
+	ingressClass := "foo-ingress"
+	tls := []netv1alpha1.IngressTLS{{
+		Hosts:      []string{"*.default.domain.com"},
+		SecretName: "secret",
+	}}
+	basicWantIngress := netv1alpha1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-route",
+			Namespace: ns,
+			Annotations: map[string]string{
+				networking.IngressClassAnnotationKey: ingressClass,
+				networking.RolloutAnnotationKey:      emptyRollout,
+			},
+			Labels: map[string]string{
+				serving.RouteLabelKey:          "test-route",
+				serving.RouteNamespaceLabelKey: ns,
+			},
+			OwnerReferences: []metav1.OwnerReference{*kmeta.NewControllerRef(r)},
+		},
+	}
+	tests := []struct {
+		name                 string
+		httpOptionAnnotation string
+		wantSpec             netv1alpha1.IngressSpec
```

Comment:
```suggestion
		wantOption             netv1alpha1.HTTPOption
```

I think we are alright to verify just expected HTTPOption around L#1113 like:

```go
                        if diff := cmp.Diff(tc.wantOption, got.Spec.HTTPOption); diff != "" {
                                t.Error("Unexpected Ingress (-want, +got):", diff)
                        }
```

Then we can remove `basicWantIngress`?

Reply:
Good point. Done